File: features/person/src/main/java/com/nivasafinance/features/person/service/PersonReadService.java
package com.nivasafinance.features.person.service;

import com.nivasafinance.common.dto.AddressData;
import com.nivasafinance.common.dto.IdentifierData;
import com.nivasafinance.features.person.dto.PersonResponse;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface PersonReadService {
    PersonResponse getPersonById(Long personId);
    PersonResponse getPersonByPrimaryMobile(String mobileNumber);
    Optional<PersonResponse> findPersonByPrimaryMobile(String mobileNumber);
    Optional<PersonResponse> findPersonByEmail(String email);
    List<PersonResponse> getPersonByMobile(String mobileNumber);
    List<AddressData> getAddresses(Long personId);
    AddressData getAddress(Long personId, String addressId);

    List<IdentifierData> getIdentifiers(Long personId);

    IdentifierData getIdentifier(Long personId, UUID identifierId);
}

File: features/person/src/main/java/com/nivasafinance/features/person/service/PersonReadService.kt
package com.nivasafinance.features.person.service

import com.nivasafinance.features.person.dto.EmploymentDetailsResponse
import com.nivasafinance.features.person.dto.PersonAddressMappingResponse
import com.nivasafinance.features.person.dto.PersonData
import com.nivasafinance.features.person.dto.PersonIdentifierResponse
import java.util.UUID

interface PersonReadService {

    // Basic person operations
    fun getPerson(id: UUID): PersonData
    fun getPersonByMobileNo(mobileNo: String): PersonData

    // Address mapping operations
    fun getPersonAddresses(personId: UUID): List<PersonAddressMappingResponse>

    // Identifier operations
    fun getPersonIdentifiers(personId: UUID): List<PersonIdentifierResponse>

    // Employment details operations
    fun getPersonEmploymentDetails(personId: UUID): EmploymentDetailsResponse?
}
